src/App.js

```diff
@@ -1,19 +1,135 @@
 import React from 'react';
+import { TodoList } from './components/TodoList';
 import './App.css';
-
+import todosFromApi from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+function getUserById(UserId) {
+  return users.find(user => user.id === UserId);
+}
+
+const preparedTodos = todosFromApi.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    taskTitle: '',
+    userId: 0,
+    errorUser: false,
+    errorTitle: false,
+  }
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+    const correctValue = name === 'userId' ? +value : value;
+
+    this.setState({ [name]: correctValue });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { userId, taskTitle } = this.state;
+
+    this.setState({
+      errorUser: false,
+      errorTitle: false,
+    });
+
+    if (userId === 0) {
+      this.setState({ errorUser: true });
+
+      return;
+    }
+
+    if (taskTitle.trim().length < 1) {
+      this.setState({ errorTitle: true });
+
+      return;
+    }
+
+    this.setState((state) => {
+      const newTask = {
+        id: Math.random(),
```

Comment:
better to use previous id + 1, think about the situation when math.random generate you the same id as you already have